test/module/irohad/model/converters/json_query_factory_test.cpp

```diff
@@ -127,6 +126,36 @@ TEST(QuerySerializerTest, DeserializeWhenUnknownType) {
   ASSERT_FALSE(res.has_value());
 }
 
+/**
+ * @given The json transaction that has valid and invalid hashes.
+ * @when Deserialize the json transaction.
+ * @then Validate the invalid hash is skipped and the only valid deserialized.
+ */
+TEST(QuerySerialzierTest, DeserializeGetTransactionsWithInvalidHash) {
+  JsonQueryFactory queryFactory;
+  iroha::hash256_t exact_size_hash{};
+  exact_size_hash[0] = 1;
+  QueryGenerator queryGenerator;
+  auto val =
+    queryGenerator.generateGetTransactions(0, "123", 0, {exact_size_hash});
+  val->signature = generateSignature(42);
+  auto json = queryFactory.serialize(val);
+  json.erase(std::remove_if(json.begin(), json.end(), [](auto c) {
+    return std::isspace(c, std::locale{});
+  }), json.end());
+  const std::string keyword = "tx_hashes\":[";
+  const auto found_pos = json.find(keyword);
+  ASSERT_NE(found_pos, std::string::npos);
+  const auto inserted_pos = found_pos + keyword.size();
+  json = std::string{json.cbegin(), json.cbegin() + inserted_pos} + "\"123\"," +
```

Comment:
If you want to change json it's better to work with it like a json, not string. Semantic of this line is also quite hard to understand

Reply:
You're right. Thank you.